docs/developer/style_guide.md

```diff
@@ -0,0 +1,579 @@
+and the order of declarations. Everything is to be private unless explicitly
+specified as public! This also makes it simple to check the code against the
+specification.
+
+
+### Layout
+
+All code is to use indentation as generated by GNU-Emacs f90-mode. Use any
+editor you like to create the code, but please ensure that whatever goes into
+the Bitbucket repository has been passed through Emacs f90-mode indentation at the
+finished stage. To assist in Emacs usage, the `ModuleTemplate.F90` file contains
+Emacs directives to set f90-mode, etc.
+
+
+### Case
+
+Use lower case for keywords and use capitalisation as given in the specification
+for all routines and variables mentioned therein. For private routines and variables
+use lower case in general. However, you may use upper case to differentiate
+words within a name, e.g. `userInputData` - as long as you are consistent
+throughout the module.
+
+Use upper case, with stripped spaces, for the value of string variables that are
+passed between routines, e.g. `run_type='MOLECULARDYNAMICS'` not
+`'MolecularDynamics`' or `'molecular dynamics'` etc. to simplify string
+comparisons.
+
+
+### Names
+
+For clarity, don't change variable name on passing to a different routine
+wherever possible. Avoid short (1 or 2 letter) variable names except for simple
+loop counters. Use new form `end subroutine name` etc. not older form
+`end subroutine`.
+
+
+### Types
+
+Do NOT use implicit typing. Similarly, you do not have to stick to the old rules
+for implicit typing - you can have integer names beginning with "`c`" etc! Do
+not give type as `double precision` - use `real(kind=dp)` instead, as `dp` is a
+constant in the Constants module and this will make porting to other
+architectures (e.g. Cray) much simpler. Remember that all machine-specific code
+stays in the Utility routines. Similarly, when declaring a constant, use
+`1.234_dp` NOT `1.234d0`.
+
+
+### Free format
+
+Use free-format f90, with a maximum line length of 132 characters.
+
+
+### Spaces
+
+Use spaces between keywords e.g. `end if` not `endif`.
+
+
+### Code cleanly
+
+Obviously, to ensure correctness and make it easier for others to maintain the
+code. If the obvious way to code a routine is inefficient, whilst the efficient
+way is obscure, then this is allowed as long as the routine is well documented,
+both internally and in this Implementation Guide, with a discussion as to the
+design decisions made.
+
+
+### Be aware of vectorisation
+
+When coding, be aware that the code is often executed upon vector architectures.
+Therefore code with a style that enables good vectorisation - e.g. don't put
+`if` statements inside nested loops wherever possible.
+
+
+## Documentation
+
+
+### Internal
+
+Each module, based upon `ModuleTemplate.F90`, is to have a header block
+describing its overall purpose and function. Note that this header block also
+contains RCS directives to log changes to the code.
```

Comment:
They still do mostly, but should probably be removed

Reply:
Agreed. We moved away from RCS ~20 years ago!